pages/in/register/index.vue

```diff
@@ -41,56 +123,274 @@ export default {
+    async handleAuthSignIn() {
+      const { method, code } = this.$route.query;
+      if (method && code) {
+        try {
+          const user = await this.handleConnectorRedirect({
+            method,
+            params: { code },
+          });
+          this.authcoreUserData = user;
+          const successLogin = await this.login({ idToken: user.idToken });
+          if (successLogin) {
+            this.redirectAfterSignIn();
+          }
+        } catch (err) {
+          const errData = err.response || err;
+          const errMessage = errData.data || errData.message || errData;
+          this.setError(errMessage, errMessage);
+          console.error(errMessage); // eslint-disable-line no-console
+        }
       }
-      logTimingEvent(this, 'RegFlow', 'CloseRegisterPageTiming', 'CloseRegisterPageTiming', value);
-      logTrackerEvent(this, 'RegFlow', 'CloseRegisterPage', 'CloseRegisterPage', value);
     },
-    logPageload() {
-      let value = 1;
-      if (window.performance) {
-        value = Math.round(performance.now());
+    async login(loginPayload = {}) {
+      let isLoginSuccessful = false;
+      const signer = this.getSigner;
+      const address = this.getAddress;
+      const payload = await signLoginMessage(signer, address);
+      const data = {
+        locale: this.getCurrentLocale,
+        platform: this.platform,
+        ...payload,
+        ...loginPayload,
+      };
+      this.signInPayload = data;
+      try {
+        await this.loginUser(this.signInPayload);
+        isLoginSuccessful = true;
+      } catch (err) {
+        if (err.response) {
+          if (err.response.status === 404) {
+            this.currentTab = TAB_OPTIONS.REGISTER;
+          }
+          isLoginSuccessful = false;
+        }
+        logTrackerEvent(this, 'RegFlow', 'LoginFail', 'LoginFail', 1);
+        // eslint-disable-next-line no-console
+        console.error(err);
+        if (this.$sentry) this.$sentry.captureException(err);
+        // this.setError((err.response || {}).data, err);
       }
-      logTimingEvent(this, 'RegFlow', 'RedirectSignUpTiming', 'RedirectSignUpTiming', value);
-      logTrackerEvent(this, 'RegFlow', 'RedirectSignUp', 'RedirectSignUp', value);
+      return isLoginSuccessful;
+    },
+
+    // Redirect
+    redirectAfterSignIn() {
+      this.currentTab = TAB_OPTIONS.REDIRECT;
+      if (!this.isOverlay) {
+        this.$nextTick(async () => {
+          if (!tryPostLoginRedirect({ route: this.$route })) {
+            // Normal case
+            await this.redirectToPreAuthPage();
+          }
+        });
+      } else {
+        this.$nextTick(async () => { await this.redirectToPreAuthPage(); });
+      }
+    },
+    redirectToPreAuthPage() {
+      this.currentTab = TAB_OPTIONS.REDIRECT;
+      const router = this.$router;
+      const route = this.$route;
+      this.doPostAuthRedirect({ router, route });
     },
+
+    // Register
+    async onCheckLikerId(likerId, check) {
```

Comment:
is `check` actually a `callback`?

Reply:
yes, call check(error) with error information if there is an error

Reply:
Please call the parameter `callback` then

Reply:
[🎨 Rename callback parameter](https://github.com/likecoin/like-co/pull/1740/commits/4021ad8d9187d3c02757a6f1aa6e2f8273aec53b)